kyo-zio/shared/src/main/scala/kyo/zios.scala

```diff
@@ -0,0 +1,68 @@
+package kyo
+
+import ZIOs.internal.*
+import core.*
+import core.internal.*
+import kyo.fibersInternal.*
+import scala.util.control.NonFatal
+import zio.Task
+import zio.ZIO
+
+opaque type ZIOs <: Fibers = Tasks & Fibers
+
+object ZIOs:
+
+    def get[E >: Nothing: Tag, A](v: ZIO[Any, E, A]): A < (Aborts[E] & ZIOs) =
+        val task = v.fold[A < Aborts[E]](e => Aborts.fail(e), a => a)
+        Tasks.suspend[A < Aborts[E], A, Aborts[E]](task, identity)
+
+    def get[A](v: ZIO[Any, Nothing, A]): A < ZIOs =
+        Tasks.suspend[A](v)
+
+    inline def get[R: zio.Tag, A](v: ZIO[R, ?, A])(using tag: Tag[Envs[R]]): A < (Envs[R] & ZIOs) =
+        compiletime.error("ZIO environments are not supported yet. Please handle them before calling this method.")
+
+    def run[T: Flat](v: T < (Aborts[Throwable] & ZIOs)): Task[T] =
+        def loop[U](v: U < ZIOs): Task[U] =
+            v match
+                case kyo: Suspend[?, ?, ?, ?] =>
+                    try
+                        if kyo.tag == Tag[IOs] then
```

Comment:
is there a reason to avoid pattern match?

```
kyo.tag match {
  case Tag[IOs]
```
etc

Reply:
It doesn't compile. I'm trying to design a "multi handler" in Kyo's core to avoid having this kind of code here and in `IOTask`.